fix: Search result are replaced by original request because it is tool ong to execute - EXO-61970 - Meeds-io/meeds#758 (#2296)

GouadriaHanen opened the pull request:
prior to this change, the Search result of users is replaced by the original request because it takes too long to retrieve the result after this change when searching people the first fetch is aborted

GouadriaHanen committed:
fix: Search result are replaced by original request because it is tool ong to execute - EXO-61970 - Meeds-io/meeds#758   (#2296)

prior to this change, the Search result of users is replaced by the original request because it takes too long to retrieve the result
after this change, the search request starts after initializing the page after retrieving the first request of user

boubaker commented on webapp/portlet/src/main/webapp/vue-apps/people-list/components/PeopleCardList.vue:
```diff
@@ -251,6 +251,15 @@ export default {
         });
 
+    },
+    waitForInitializing() {
+      window.setTimeout(() => {
+        if (this.initialized) {
+          this.searchPeople();
+        } else {
+          this.waitForInitializing();
+        }
+      }, 50);
```
```suggestion
```

Add a watch on `initialized` instead :

```javascript
watch: {
  initialized() {
    if (this.initialized && this.keyword?.length) {
          this.searchPeople();
    }
  }
},
```

boubaker replied:
BTW, it would be more elegant IMO to abort first fetch and execute right a way the user query by using :
https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort

```javascript
  if (this.controller) {
    this.controller.abort();
  }
  this.controller = new AbortController();,
  ...
  // Pseudo algorithm, this will allow to have `this.signal` variable in `getUsers` method without adding signal a argument which will break the API signature
  this.$userService.getUsers.apply({singal: this.controller.signal}, this.keyword, this.offset...)
  .finally(() => {
...
    this.controller = null;
  });
....
```

GouadriaHanen committed:
fix: Search result are replaced by original request because it is tool ong to execute - EXO-61970 - Meeds-io/meeds#758 (#2358)

prior to this change, the Search result of users is replaced by the
original request because it takes too long to retrieve the result after
this change when searching people the first fetch is aborted
